src/struct_info.rs

```diff
@@ -574,11 +580,31 @@ impl CommonDeclarationSettings {
     }
 }
 
+/// Setting of the `into` argument.
+#[derive(Debug)]
+pub enum IntoSetting {
+    /// Setting was not set.
+    Unset,
+    /// Setting was set without a value.
+    Set,
+    /// Setting was set with a specific value (type).
+    Type(syn::ExprPath),
+}
+
+impl Default for IntoSetting {
+    fn default() -> Self {
+        Self::Unset
+    }
+}
+
 #[derive(Debug, Default)]
 pub struct TypeBuilderAttr {
     /// Whether to show docs for the `TypeBuilder` type (rather than hiding them).
     pub doc: bool,
 
+    /// Whether to convert the builded type into another while finishing the build.
+    pub into: IntoSetting,
```

Comment:
I think this belongs inside `build_method` settings, but given how they all currently use `CommonDeclarationSettings` it might be easier to keep it here for now, and I'll move it to `builder_method` after the merge (there is some work to be done with the attribute type, that I prefer to do myself)

Reply:
Yeah I had my trouble with that, too. But I was also thinking, it might be symbolically still fine this way (as the whole builder will build into something different) and would allow to possibly modify the `builder()` method's visibility as well.